src/Accessor/DefaultAccessorStrategy.php

```diff
@@ -20,21 +20,76 @@
 
 namespace JMS\Serializer\Accessor;
 
+use JMS\Serializer\Exception\ExpressionLanguageRequiredException;
+use JMS\Serializer\Exception\LogicException;
+use JMS\Serializer\Expression\ExpressionEvaluatorInterface;
+use JMS\Serializer\Metadata\ExpressionPropertyMetadata;
 use JMS\Serializer\Metadata\PropertyMetadata;
+use JMS\Serializer\Metadata\StaticPropertyMetadata;
+use JMS\Serializer\Metadata\VirtualPropertyMetadata;
 
 /**
  * @author Asmir Mustafic <goetas@gmail.com>
  */
 final class DefaultAccessorStrategy implements AccessorStrategyInterface
 {
+    private $readAccessors = array();
+    private $writeAccessors = array();
```

Comment:
Inho one shared accessor is enough, no? They are bound to type only.

Reply:
The body of the closure is differnet, 

```php
function ($o, $name, $value) {
   $o->$name = $value;
}
```
vs
```php
function ($o, $name {
   return $o->$name;
}
```
so cant be done